scripts/exregional_run_prepstart.sh

```diff
@@ -136,8 +167,8 @@ else
     else
       n=$((n + ${DA_CYCLE_INTERV}))
       YYYYMMDDHHmInterv=$( date +%Y%m%d%H -d "${START_DATE} ${n} hours ago" )
-      bkpath=${fg_root}/${YYYYMMDDHHmInterv}/fcst_fv3lam/RESTART  # cycling, use background from RESTART
-      if [ ${n} -eq ${FCST_LEN_HRS} ]; then
+      bkpath=${fg_root}/${YYYYMMDDHHmInterv}/${fcst_dir_name}/RESTART  # cycling, use background from RESTART
+      if [ ${n} -eq ${FCST_LEN_HRS_SPINUP} ] && [ ${cycle_type} -eq 1 ]; then
```

Comment:
The former version does not work well  because it is very hard to know the forecast length in previous cycles.
This  part of check is more useful for spin-up cycles because spin-up cycle always use the same forecast length and only run a very shout forecast. That is why we only use this check for spin-up cycle now.